libass/ass_parse.c

```diff
@@ -120,8 +120,9 @@ void ass_update_font(RenderContext *state)
  */
 static inline int32_t dtoi32(double val)
 {
-    if (isnan(val) || val <= INT32_MIN || val >= INT32_MAX + 1LL)
+    if (!(val > INT32_MIN - 1LL && val < INT32_MAX + 1LL))
```

Comment:
Any particular reason to add the `- 1LL`?

Also, maybe invert the conditional control flow to avoid the slightly-confusing negation?

Reply:
Hmm, I'd hoped the 1LL might provide a better hint to the compiler here, but it looks like it doesn't. I'll remove.

Code generated is identical regardless of the negation, so sure, I'll swap that as well.

Wondering if a call to `__builtin_ia32_cvttsd2si` is worthwhile here (it's the only way I've found to convince the compiler to actually just do the conversion unchecked on x86 without invoking UB in the out-of-range case), but I'm a bit concerned that it might constrain optimizations that could've otherwise skipped the cast altogether.